lib/Gitlab/HttpClient/Builder.php

```diff
@@ -168,16 +168,20 @@ public function addPlugin(Plugin $plugin)
      *
      * @param string $fqcn
      *
-     * @return void
+     * @return Plugin|null
      */
     public function removePlugin($fqcn)
     {
+        $plugin = null;
+
         foreach ($this->plugins as $idx => $plugin) {
```

Comment:
How about we store plugins as a map from FQCN to instance instead?

Not sure if adding a plugin twice is ever needed. If so, I can change this method to return an array of removed plugins instead.